app/src/androidTest/java/org/kiwix/kiwixmobile/DownloadTest.kt

```diff
@@ -0,0 +1,127 @@
+import com.adevinta.android.barista.interaction.BaristaClickInteractions.clickOn
+import com.adevinta.android.barista.interaction.BaristaMenuClickInteractions.clickMenu
+import com.adevinta.android.barista.interaction.BaristaSleepInteractions
+import com.adevinta.android.barista.interaction.BaristaSwipeRefreshInteractions.refresh
+import junit.framework.Assert
+import org.junit.After
+import org.junit.Before
+import org.junit.BeforeClass
+import org.junit.Ignore
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.kiwix.kiwixmobile.main.KiwixMainActivity
+import org.kiwix.kiwixmobile.testutils.TestUtils
+import org.kiwix.kiwixmobile.utils.KiwixIdlingResource.Companion.getInstance
+import org.kiwix.kiwixmobile.utils.StandardActions
+import java.util.concurrent.TimeUnit
+
+@LargeTest
+@RunWith(AndroidJUnit4::class)
+class DownloadTest {
+  @Rule
+  var activityTestRule = ActivityTestRule(
+    KiwixMainActivity::class.java
+  )
+
+  @Rule
+  var readPermissionRule: GrantPermissionRule =
+    GrantPermissionRule.grant(Manifest.permission.READ_EXTERNAL_STORAGE)
+
+  @Rule
+  var writePermissionRule: GrantPermissionRule =
+    GrantPermissionRule.grant(Manifest.permission.WRITE_EXTERNAL_STORAGE)
+
+  @Before fun setUp() {
+    // default
+  }
+
+  @Test @Ignore("Broken in 2.5") // TODO: Fix in 3.0
+  fun downloadTest() {
+    BaristaSleepInteractions.sleep(TestUtils.TEST_PAUSE_MS.toLong())
+    clickMenu(TestUtils.getResourceString(R.string.library))
+    clickOn(R.string.local_zims)
+    TestUtils.allowPermissionsIfNeeded()
+    StandardActions.deleteZimIfExists("ray_charles", R.id.zimfilelist)
+    clickOn(R.string.remote_zims)
+    TestUtils.captureAndSaveScreenshot("Before-checking-for-ZimManager-Main-Activity")
+    TestUtils.captureAndSaveScreenshot("After-the-check-completed")
+    BaristaSleepInteractions.sleep(TestUtils.TEST_PAUSE_MS.toLong())
+    try {
+      Espresso.onData(TestUtils.withContent("ray_charles"))
+        .inAdapterView(ViewMatchers.withId(R.id.libraryList))
+    } catch (e: Exception) {
+      Assert.fail(
+        """
+  Couldn't find downloaded file 'ray_charles'
+  
+  Original Exception:
+  ${e.localizedMessage}
+  
+  
+        """.trimIndent()
+      )
+    }
+    StandardActions.deleteZimIfExists("ray_charles", R.id.libraryList)
+    assertDisplayed(R.string.local_zims)
+    clickOn(R.string.local_zims)
+    try {
+      refresh(R.id.zim_swiperefresh)
+    } catch (e: RuntimeException) {
+      Log.w(KIWIX_DOWNLOAD_TEST, "Failed to refresh ZIM list: " + e.localizedMessage)
+    }
+
+/*
+Commented out the following as it uses another Activity.
+TODO Once we find a good way to run cross-activity re-implement
+this functionality in the tests.
+
+    onData(withContent("ray_charles")).inAdapterView(withId(R.id.zimfilelist)).perform(click());
+    openContextualActionModeOverflowMenu();
+    onView(withText("Get Content")).perform(click());
```

Comment:
[nit] The documentation looks odd in the idea. It should start with the spaces or horizontally aligned with the function's name.

Reply:
Done